front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -66,18 +66,22 @@ async function handler(
     });
   }
 
+  // No role under "builder" can create a managed data source.
+  // We perform a more details check below for each provider,
+  // but this is a first line of defense.
+  if (!auth.isBuilder()) {
```

Comment:
For coherence with other endpoints let's just add || !auth.isBuilder() in the previous condition. 404 when non auth is fine and even desirable :+1: